preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py

```diff
@@ -47,10 +47,30 @@ def invalid_value_annotation(input_datum, output_field, value_type) -> Processin
 
 
 def valid_authors(authors: str) -> bool:
-    pattern = r"^([a-zA-Z\s\.\-\']+,[a-zA-Z\s\.\-\']*;)+"
+    pattern = r"^([a-zA-Z\s\.\-\']+,[a-zA-Z\s\.\-\']*;)*([a-zA-Z\s\.\-\']+,[a-zA-Z\s\.\-\']*;*)$"
     return re.match(pattern, authors) is not None
 
 
+def warn_potentially_invalid_authors(authors: str) -> bool:
+    authors_split = re.split(r"[,\s]+", authors)
+    return bool(";" not in authors and len(authors_split) > 3)
```

Comment:
```suggestion
def warn_potentially_invalid_authors(authors: str) -> bool:
    if ";" in authors:
        return False
    
    if "," not in authors:
        return False
    
    parts = authors.split(",", 1)
    before_comma = re.split(r"\s+", parts[0].strip())
    after_comma = re.split(r"\s+", parts[1].strip())
    
    return len(before_comma) >= 2 and len(after_comma) >= 2
```

Reply:
I'm not sure we need this change. If `,` is not in `authors` the string will be invalid and will be caught by the valid_authors check/function. 

I'm fine with changing this but think the function I have is a bit simpler. Potentially we warn a bit more with my function e.g. if someone has three first names or last names - but as this is anyways very rare I think its fine to do that (might also indicate an error) - especially as the warning doesn't block submission.

Reply:
(sorry for the deleting and adding back I started a review by accident and just wanted to comment)

Reply:
> If `,` is not in authors the string will be invalid and will be caught by the valid_authors check/function.

I would suggest we may want to remove that check because I think `the RECOVERY Consortium` and similar may be quite common author names.

I don't feel strongly about the other aspects

Reply:
Hmm this is an interesting Q, thanks! I agree I was not understanding the requirements.

I see that for e.g. https://www.ncbi.nlm.nih.gov/nuccore/OZ013273.1 the submitters have hacked it as you describe: `Danish Covid-19 Genome,Consortium.`

Reply:
I think I still feel like ideal behaviour for Pathoplexus might be to allow a more natural way of submitting these ,and for us to do some translation for submission to ENA. But if we're not providing a degradation over the current system this feels like something we can push to the future. Thanks for explaining.